lib/ocpp/common/websocket/websocket_tls_tpm.cpp

```diff
@@ -276,29 +229,27 @@ void WebsocketTlsTPM::tls_init() {
     SSL_CTX_set_ecdh_auto(ctx, 1);
 
     if (this->connection_options.security_profile == 3) {
-        const char* path_key = nullptr;
-        const char* path_chain = nullptr;
-
-        const auto certificate_key_pair =
-            this->evse_security->get_key_pair(CertificateSigningUseEnum::ChargingStationCertificate);
-
-        if (!certificate_key_pair.has_value()) {
-            EVLOG_AND_THROW(std::runtime_error(
-                "Connecting with security profile 3 but no client side certificate is present or valid"));
+        if ((path_chain == nullptr) || (path_key == nullptr)) {
+            EVLOG_error << "Cert chain: " << path_chain << " key: " << path_key;
+            EVLOG_AND_THROW(std::runtime_error("No certificate and key for SSL"));
         }
 
-        path_chain = certificate_key_pair.value().certificate_path.c_str();
-        path_key = certificate_key_pair.value().key_path.c_str();
-
         if (1 != SSL_CTX_use_certificate_chain_file(ctx, path_chain)) {
+            ERR_print_errors_fp(stderr);
             EVLOG_AND_THROW(std::runtime_error("Could not use client certificate file within SSL context"));
         }
 
+        OpenSSLProvider provider;
```

Comment:
I couldn't find any usage for this. Isn't it used outside the function already?

Reply:
it adds no value any more - removing